### PR TITLE
Add support for AWS::Scheduler::Schedule and AWS::Scheduler::ScheduleGroup

### LINKED ISSUE #8777: enhancement request: AWS::Scheduler::Schedule and AWS::Scheduler::ScheduleGroup
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Enhancement description

https://github.com/localstack/localstack/pull/8754 added EventBridge Scheduler support however LocalStack is missing support for resource types: `AWS::Scheduler::Schedule` and `AWS::Scheduler::ScheduleGroup`

Follow up on https://github.com/localstack/localstack/issues/7268

### 🧑‍💻 Implementation

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-scheduler-schedule.html

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-scheduler-schedulegroup.html

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by pinzon**
## Motivation
The CRUD API operations for Scheduler::Schedule and Scheduler::ScheduleGroup were recently added to LocalStack so the support with CFn would be expected.
Closes #8777
## Changes
- Resource Provider for AWS::Scheduler::Schedule 
- Resource Provider for AWS::Scheduler::ScheduleGroup
## Testing
- Validates the deployment of a stack with the resources

### CODE REVIEW

File: localstack/services/scheduler/resource_providers/aws_scheduler_schedulegroup.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+        Delete a resource
+
+        IAM permissions required:
+          - scheduler:DeleteScheduleGroup
+          - scheduler:GetScheduleGroup
+          - scheduler:DeleteSchedule
+        """
+        model = request.desired_state
+        request.aws_client_factory.scheduler.delete_schedule_group(Name=model["Name"])
+        return ProgressEvent(status=OperationStatus.SUCCESS, resource_model={})
```
Comment (dominikschubert) [inline localstack/services/scheduler/resource_providers/aws_scheduler_schedulegroup.py:110]: a bit of a nit but it seems CFn actually first deletes all schedules here before deleting the group :thinking:

File: tests/aws/services/cloudformation/resource_providers/scheduler/test_scheduler.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    snapshot.add_transformer(snapshot.transform.cloudformation_api())
+    schedule = aws_client.cloudformation.describe_stack_resource(
+        StackName=stack.stack_name, LogicalResourceId="MySchedule"
+    )["StackResourceDetail"]
+    snapshot.match("Schedule", schedule)
+
+    group = aws_client.cloudformation.describe_stack_resource(
+        StackName=stack.stack_name, LogicalResourceId="MyScheduleGroup"
+    )["StackResourceDetail"]
+    snapshot.match("Group", group)
```
Comment (dominikschubert) [inline tests/aws/services/cloudformation/resource_providers/scheduler/test_scheduler.py:25]: would be great if we could add some specific API describe calls as well, i.e. get_schedule / get_schedule_group

**Review by dominikschubert — APPROVED** [review]
minor nits, other than that LGTM!

File: localstack/services/scheduler/resource_providers/aws_scheduler_schedulegroup.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+        """
+        Delete a resource
+
+        IAM permissions required:
+          - scheduler:DeleteScheduleGroup
+          - scheduler:GetScheduleGroup
+          - scheduler:DeleteSchedule
+        """
+        model = request.desired_state
+        schedules = request.aws_client_factory.scheduler.list_schedules(
```
Comment (dominikschubert) [inline localstack/services/scheduler/resource_providers/aws_scheduler_schedulegroup.py:109]: I'm actually confused by the IAM permissions here. How are they getting the schedule to delete if they don't have a `scheduler:ListSchedules` permission? :thinking:

Comment (pinzon) [reply localstack/services/scheduler/resource_providers/aws_scheduler_schedulegroup.py]: [The docs of from the boto3 operation](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/scheduler/client/delete_schedule_group.html) confirm that the api service will delete the schedules if the schedule group is deleted. it's not something that the resource provider should be responsible then.

**Pull request merged**